department-of-veterans-affairs/abd-vro
# MCP-2096 Claim Info Endpoint

## What was the problem?
 We need to search claims using a particular claimSubmissionId or  a veteranIcn.  We also need to see the all the processed claims stored in the database using pagination. 

- [x] Find claim with claimSubmissionId
- [x] Find claims with veteranIcn
- [x] Find all claims using pagination
- [x] Remove fetch-claims endpoint

Associated tickets or Slack threads:

- [MCP-2096](https://amida.atlassian.net/browse/MCP-2096)

## How does this fix it?[^1]
Creates endpoints which will display claims based on the input parameters.

## How to test this PR
- Step 1: Run full health assessment with claimSubmissionId = 1234 
- Step 2:  Execute the claim-info endpoint using claimSubmissionId 1234
- Step 3: Execute the claim-info-for-veteran endpoint using veteranIcn = 
- Step 4: run full-health-assessment a few times with different claim submission ids each time.
- Step 5: Execute the claim-info-with-pagination endpoint and specify the offset as 0 and the pageSize as 10.
- Step 6: Verify all the claims are returned successfully.


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
